core/controllers/base.py

```diff
@@ -578,3 +578,17 @@ def is_csrf_token_valid(cls, user_id, token):
             return False
         except Exception:
             return False
+
+
+class CsrfTokenHandler(BaseHandler):
+    """Handles sending CSRF tokens to the frontend."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    REQUIRE_PAYLOAD_CSRF_CHECK = False
+
+    def get(self):
```

Comment:
Just to check, should this handler require login? There is an assumption that self.user_id exists.

If it's OK for self.user_id to be None, maybe we should handle that explicitly.

Reply:
OK, thanks for checking. (Though if you could update [this](https://github.com/oppia/oppia/blob/e69b57106e42f5b7f80e8556b3bd62713f4fadf2/core/controllers/base.py#L505) and similar docstrings to say str|None, that would be great.)